src/librustc_resolve/diagnostics.rs

```diff
@@ -1029,6 +1029,32 @@ struct Bar2; // ok!
 ```
 "##,
 
+E0429: r##"
+To import a namespace itself in addition to some of its members, the `self`
```

Comment:
This paragraph should just be the description of the error, not how to fix it. Put this paragraph just after the erroneous code example and replace it here by the error description.